### PR TITLE
lib/promscrape: add relabel configs to `global` section

### PR DESCRIPTION
**Pull request by f41gh7**
This commit adds `metric_relabel_configs` and `relabel_configs` fields into the `global` section of scrape configuration file.
 New fields are used as global relabeling rules for the scrape targets.
 These relabel configs are prepended to the target relabel configs.
This feature is useful to:
* apply global rules to __meta labels from service discovery targets.
* drop noisy labels during scrapping.
* mutate labels without affecting metrics ingested via any of push protocols. Related issue https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6966

### CODE REVIEW

**Review by AndrewChubatiuk — APPROVED** [review]

File: docs/vmagent.md [hunk 1]
```diff
@@ -573,6 +573,16 @@ adds `exported_` prefix to these metric names, so they don't clash with automati
 VictoriaMetrics components support [Prometheus-compatible relabeling](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config)
 with [additional enhancements](#relabeling-enhancements). The relabeling can be defined in the following places processed by `vmagent`:
 
+* At the `global -> relabel_configs` section in `-promscrape.config` file.
+  This relabeling is used for modifying labels in discovered targets and for dropping unneeded targets.
+  Configuration from global section will be prepended to the `relabel_config` of targets from `scrape_config` section.
+  See [relabeling cookbook](https://docs.victoriametrics.com/relabeling/) for details.
+
```
Comment (jiekun) [inline docs/vmagent.md:580]:
```suggestion
_Available from [TODO](https://docs.victoriametrics.com/changelog/#TODO) version._

```

File: docs/vmagent.md [hunk 2]
```diff
@@ -573,6 +573,16 @@ adds `exported_` prefix to these metric names, so they don't clash with automati
...
+* At the `global -> metric_relabel_configs` section in `-promscrape.config` file.
+  This relabeling is used for modifying labels in scraped metrics and for dropping unneeded metrics.
+  Configuration from global section will be prepended to the `metric_relabel_config` of targets from `scrape_config` section.
+  See [relabeling cookbook](https://docs.victoriametrics.com/relabeling/) for details.
+
```
Comment (jiekun) [inline docs/vmagent.md:585]:
```suggestion
_Available from [TODO](https://docs.victoriametrics.com/changelog/#TODO) version._

```

**Review by jiekun — APPROVED** [review]
LGTM. Some optional changes for the doc. Please apply as per your preference.

File: lib/promscrape/config.go [hunk 3]
```diff
@@ -903,11 +905,23 @@ func getScrapeWorkConfig(sc *ScrapeConfig, baseDir string, globalCfg *GlobalConf
...
+		rcs = append(rcs, globalCfg.RelabelConfigs...)
+		rcs = append(rcs, sc.RelabelConfigs...)
+	}
+	relabelConfigs, err := promrelabel.ParseRelabelConfigs(rcs)
 	if err != nil {
 		return nil, fmt.Errorf("cannot parse `relabel_configs` for `job_name` %q: %w", jobName, err)
 	}
-	metricRelabelConfigs, err := promrelabel.ParseRelabelConfigs(sc.MetricRelabelConfigs)
+	mrcs := sc.MetricRelabelConfigs
+	if len(globalCfg.MetricRelabelConfigs) > 0 {
```
Comment (f41gh7) [inline lib/promscrape/config.go:919]: @hagen1778  Open question to me when we should apply relabeling:

1) before actual relabeling defined at job (current version). Allows to inject default values, normalize labels.
2) after job relabeling. It's better when you want to "enforce" labels.

Comment (hagen1778) [reply lib/promscrape/config.go:934]: We should keep behavior as is: going from global relabeling rules to per-job relabeling rules.

**Review by hagen1778 — APPROVED** [review]
LGTM

**Pull request merged**